External/FEXCore/Source/Interface/Core/JIT/x86_64/JIT.cpp

```diff
@@ -836,18 +836,31 @@ void JITCore::CreateCustomDispatch(FEXCore::Core::InternalThreadState *Thread) {
   mov(qword [STATE + offsetof(FEXCore::Core::ThreadState, ReturningStackLocation)], rsp);
 
   Label LoopTop;
+  Label FullLookup;
   Label NoBlock;
   Label ThreadPauseHandler{};
 
   L(LoopTop);
   AbsoluteLoopTopAddress = getCurr<uint64_t>();
-
   {
-    mov(r13, Thread->BlockCache->GetPagePointer());
-
     // Load our RIP
     mov(rdx, qword [STATE + offsetof(FEXCore::Core::CPUState, rip)]);
 
+    // L1 Cache
+    mov(r13, Thread->BlockCache->GetL1Pointer());
+    mov(rax, rdx);
+
+    and_(rax, 1 * 1024 * 1024 - 1);
```

Comment:
Don't use hard coded L1C size.
Pull a constant from the BlockCache class instead

Reply:
Done